fairlearn/metrics/_annotated_metric_function.py

```diff
@@ -90,19 +93,16 @@ def __call__(self, df: pd.DataFrame):
 
         The second issue is coping with when users have passed in a 2D array as
         a named argument (especially, `y_true` or `y_pred`).
-        For this reason, we perform some extra list-washing, to make sure the
+        For this reason, we perform an extra `np.stack` operation, to make sure the
         expected types are passed to the underlying metric function.
         """
-        args = []
-        for arg_name in self.postional_argument_names:
-            # Need to convert to list first in case we have 2D arrays
-            args.append(np.asarray(list(df[arg_name])))
-
-        kwargs = dict()
-        for func_arg_name, data_arg_name in self.kw_argument_mapping.items():
-            # Need to convert to list first in case we have 2D arrays
-            kwargs[func_arg_name] = np.asarray(list(df[data_arg_name]))
-
-        result = self.func(*args, **kwargs)
-
-        return result
+        df_nw = nw.from_native(df, eager_only=True, pass_through=False)
+        args = [
+            np.stack(df_nw.get_column(arg_name).to_numpy())
```

Comment:
Please let's make sure that there are no edge cases we could hit with `np.stack` that would otherwise be covered when converting to list.

I hear the disappointment in the "we perform some extra list-washing [...]" and tried to come up with something avoiding using a list call 😁